config/initializers/devise.rb

```diff
@@ -274,6 +274,30 @@
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', scope: 'user,public_repo'
 
+  # HPI OpenID Connect configuration
+  config.omniauth :openid_connect,
+                  name: :openid_connect,
+                  scope: %i[openid email profile],
+                  response_type: :code,
+                  client_options: {
+                    port: 443,
+                    scheme: "https",
+                    host: "oidc.hpi.de",
+                    # Instead of env vars, could also use Rails credentials store
+                    # env vars are set on deployed Heroku instance, default to HPI OpenID client setup for local dev
+                    # Requires server to be running on port 3000, as that is also set on the remote OIDC config (and is checked)
+                    identifier: ENV.fetch(
+                      "OPENID_CONNECT_CLIENT_ID"
+                    ),
+                    secret: ENV.fetch(
+                      "OPENID_CONNECT_CLIENT_SECRET"
+                    ),
+                    redirect_uri: "http://localhost:3000/users/auth/openid_connect/callback",
```

Comment:
Would this also work in production, where `localhost:3000` isn't available? Or in development, if somebody starts the server on a non-3000 port?

Removing the host seems less brittle to me:

```suggestion
                    redirect_uri: "/users/auth/openid_connect/callback",
```

I'm not familiar with OmniAuth nor OIDC, so maybe this works via some magic or is even required 🤷‍♀️

Reply:
Not possible, OIDC requires the full path for the redirect URI. Although hardcoding to localhost is not a great idea either. Suggestion would be to append the path to a base URL which is configurable via env variables, however default to localhost if the varaible is not set.

Reply:
Adapted to work in a production environment w/ a local-only OIDC client set as a fallback.